### DISCUSSION TITLE
Duplicate field : Foreign Keys is disabled

### DISCUSSION THREAD

**Discussion by VaZark** [directus/directus#11673] [Archive]
Why can't the _Duplicate field_ action be applied over foreign key fields? As far as I understand, the only change that'll be necessary is that the relation name has to be regenerated.

**Comment by rijkvanzanten** [comment]
The main reason is just that: the added complexity of duplicating relational fields hasn't been implemented yet. Copying a many-to-one is (like you said) easy enough, as it's just the single relationship with a new name. Copying any of the other relational types however becomes pretty complicated, as you effectively have to go through the same relational configuration flow as you would for a new field.

The real solution here could be that the "duplicate" action effectively pulls up the field-create flow, but with the existing configuration from the previous field pre-filled (including relationships). That way, you can still apply any changes required (like what junction tables are used etc) before having the field duplicated, which in turn would allow for relational fields to be more easily duplicated.

**Reply by rijkvanzanten** [reply]
The only thing with this approach is that we'll have to figure out how to efficiently allow this to-be-created field to be saved into a different collection from the current one, without being confusing for the end user. That's a solvable design challenge though 🙂
